lib/app/admin/data-reset/remove.js

```diff
@@ -0,0 +1,77 @@
+
+var q = require('q');
+var async = require('async');
+var DEFAULT_ITEM_LIST = require('./default_item_type_list');
+
+/**
+ * removes all data-reset in the solution
+ * @param {Object} mediator - mediator object used for communication
+ * @returns {Promise} Promise, resolved if all data-reset are successfully removed.
+ */
+var removeData = function removeData(mediator) {
+  var deferred = q.defer();
+  console.log('>>>>remove start');
+  async.eachSeries(DEFAULT_ITEM_LIST, function(item, eachSeriesCb) {
+    // remove all items of a type
+    removeAllItemsForType(mediator, item)
+      .then(eachSeriesCb)
```

Comment:
If `eachSeriesCb` throws an error the `then` will catch it, and this means the `catch` will be invoked, thus invoking the same function and therefore same error twice. Depending on how the promises are setup this means the error is completely swallowed, bubbles up adn crashes the process, or goes to a gloablRejectionHandler (node 4+ and some promise libs I think).

This is very unlikely since async `eachSeriesCb` probably won't throw, but who knows really.

You don't want this.